### PR TITLE
[RQ-677] feat: password less email sign in with magic link

### PR DESCRIPTION
**Pull request by nsrCodes**
https://github.com/requestly/requestly/assets/57226514/b2356fa8-7293-4560-8910-bca0dd96f3b4

### CODE REVIEW

File: app/src/components/misc/SignInViaEmailLink/SignInViaEmailLink.js [hunk 1]
```diff
@@ -1,115 +1,145 @@
...
+            id="SignInViaEmailLinkLoginBtn"
+            type="primary"
+            size="large"
+            onClick={(e) => {
+              e.preventDefault();
+              handleLogin();
+            }}
+            disabled={userEmail && isProcessing}
+          >
+            {userEmail && isProcessing ? <FaSpinner className="icon-spin mr-2" /> : "Login"}
```
Comment (nafees87n) [inline app/src/components/misc/SignInViaEmailLink/SignInViaEmailLink.js:117]: We can use the loading prop in `RQButton`

File: app/src/components/misc/SignInViaEmailLink/index.css [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  width: 80%;
+}
+
+.container {
+  margin: 20px 50px;
+  padding: 0 50px;
+}
+.mb-20 {
+  margin-bottom: 20px;
+}
```
Comment (nafees87n) [inline app/src/components/misc/SignInViaEmailLink/index.css:14]: - These class names are generic. If you require class for a specific component, please use specific class names for the component.

- `mb-20` can be setup in `base.css` if you require.
- Avoid use of `!important`

File: app/src/components/authentication/AuthForm/GenerateLoginLinkButton.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        });
+    }
+  }, [callback, email, startResetTimerAndAnimation]);
+
+  return isWaiting ? (
+    <RQButton
+      title="please check your email for the login link"
+      className="form-elements-margin w-full"
+      disabled
+      style={{
```
Comment (nafees87n) [inline app/src/components/authentication/AuthForm/GenerateLoginLinkButton.jsx:74]: Shouldn't use inline style

File: app/src/components/authentication/AuthForm/GenerateLoginLinkButton.jsx [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    requestAnimationFrame(updateProgress);
+  }, [timer, timerEndCallback]);
+
+  const handleBtnClick = useCallback(() => {
+    if (!email || !isEmailValid(email)) {
+      toast.warn("Please enter a valid email");
+    } else {
+      startResetTimerAndAnimation();
+      console.log(sendEmailLinkForSignin);
```
Comment (nafees87n) [inline app/src/components/authentication/AuthForm/GenerateLoginLinkButton.jsx:57]: missed an debug log

**Review by nafees87n — CHANGES_REQUESTED** [review]

File: app/src/components/misc/SignInViaEmailLink/SignInViaEmailLink.js [hunk 5]
```diff
@@ -1,115 +1,143 @@
...
+            id="SignInViaEmailLinkLoginBtn"
+            type="primary"
+            size="large"
+            onClick={(e) => {
+              e.preventDefault();
+              handleLogin();
+            }}
+            loading={userEmail && isProcessing}
+          >
+            LOGIN
```
Comment (nafees87n) [inline app/src/components/misc/SignInViaEmailLink/SignInViaEmailLink.js:115]:
```suggestion
            Login
```

**Review by nafees87n — APPROVED** [review]
@nsrCodes There is lint warning about unused components in `src/components/authentication/AuthForm/index.js`. Please fix that before merging.

**Comment by nsrCodes** [comment]
updated UI:
Screenshot 2023-09-15 at 9 55 08 AM
Screenshot 2023-09-15 at 9 55 19 AM
Screenshot 2023-09-15 at 9 55 28 AM

**Comment by RuntimeTerror10** [comment]
Magic link sign in button not visible in workspace onboarding login screen

Screenshot 2023-09-15 at 3 14 54 PM

**Review by RuntimeTerror10 — APPROVED** [review]

**Comment by nsrCodes** [comment]
updated auth modals
Screenshot 2023-09-22 at 9 48 21 AM
Screenshot 2023-09-22 at 9 48 34 AM

**Pull request merged**